mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulContestsScreen.kt

```diff
@@ -34,7 +40,27 @@ data class TournamentAdapter(val tournament: Tournament, val currentUser: Authen
         null
       }
   // TODO: Change to tournament.status when added.
-  override val status: Status = Status.Done
+  override val status: Status = Status.Ongoing(roundedDuration(tournament.creationTime))
+}
+
+/**
+ * Obtains the elapsed duration from the startTime and rounds it to the closest unit of time
+ * (seconds minimum).
+ *
+ * @param startTime creation time in milliseconds to obtain the elapsed rounded duration.
+ */
+private fun roundedDuration(startTime: Long): Duration {
+  val duration = (System.currentTimeMillis() - startTime).milliseconds
+
+  if (duration >= 1.days) {
+    return duration.inWholeDays.days
+  } else if (duration >= 1.hours) {
+    return duration.inWholeHours.hours
+  } else if (duration >= 1.minutes) {
+    return duration.inWholeMinutes.minutes
+  }
+
+  return duration.inWholeSeconds.seconds
```

Comment:
Question : what would you think of letting the facade provide the duration since the tournament was open ? These computations seem like something that's domain specific.

Reply:
This was an approach I had actually, but it would require adding the facade as an additional argument to the TournamentAdapter correct?

Reply:
Not really; you could provide the duration in the tournament objects returned by the facade.